worlds/yoshisisland/__init__.py

```diff
@@ -266,25 +260,24 @@ def get_excluded_items(self) -> Set[str]:
 
     def create_item_with_correct_settings(self, player: int, name: str) -> Item:
         data = item_table[name]
-        classification = self.item_classifications[data.classification]
-        item = Item(name, classification, data.code, player)
+        item = Item(name, data.classification, data.code, player)
 
         if not item.advancement:
             return item
 
-        if name == 'Car Morph' and self.options.stage_logic.value != 0:
+        if name == 'Car Morph' and not self.options.stage_logic:
```

Comment:
This seems like a mistake? You reversed what you're looking for regarding stage_logic

Reply:
So it was